Support OVN-IC on DPU

wizhaoredhat opened the pull request.

wizhaoredhat committed:
Update apimachinery

Co-authored-by: Riccardo Ravaioli <rravaiol@redhat.com>
Signed-off-by: William Zhao <wizhao@redhat.com>

wizhaoredhat committed:
Support OVN-IC on DPU

Co-authored-by: Riccardo Ravaioli <rravaiol@redhat.com>
Signed-off-by: William Zhao <wizhao@redhat.com>

wizhaoredhat commented:
/cc @ricky-rav @bn222 @SalDaniele

ricky-rav commented on controllers/config_controller.go:
```diff
@@ -245,11 +270,32 @@ func (r *DpuClusterConfigReconciler) syncOvnkubeDaemonSet(ctx context.Context, c
+	// - name: KUBE_RBAC_PROXY_IMAGE
+	//   value: "quay.io/openshift/origin-kube-rbac-proxy:latest"
+	// https://github.com/openshift/cluster-network-operator/blob/master/manifests/0000_70_cluster-network-operator_03_deployment.yaml#L69-L70
+	data.Data["KubeRBACProxyImage"] = "registry.redhat.io/openshift4/ose-kube-rbac-proxy" // os.Getenv("KUBE_RBAC_PROXY_IMAGE")
```
we can remove this comment of mine :)

wizhaoredhat replied:
Done

ricky-rav commented on controllers/config_controller.go:
```diff
@@ -245,11 +270,32 @@ func (r *DpuClusterConfigReconciler) syncOvnkubeDaemonSet(ctx context.Context, c
+		// Less resource constrained clusters can use multiple threads
+		// in northd to improve network operation latency at the cost
+		// of a bit of CPU.
+		data.Data["NorthdThreads"] = 4
```
we went back to having 1 thread only for northd when deployed with ovn IC:
https://github.com/openshift/cluster-network-operator/commit/703ccb516dc5eef5a5d108f2458e31c54f258428

wizhaoredhat replied:
Thanks for sharing =)

wizhaoredhat committed:
Revised based on feedback

wizhaoredhat committed:
Join ovnkube-controller and ovnkube-node container for multizone setup

Taken from https://github.com/openshift/cluster-network-operator/pull/1971
to optimize ovn-k.

Signed-off-by: William Zhao <wizhao@redhat.com>
